devops/jobs/ConfigurationWatcher.groovy

```diff
@@ -0,0 +1,154 @@
+
+    Variables consumed from the EXTRA_VARS input to your seed job in addition
+    to those listed in the seed job.
+    APPS_TO_CONFIG: a dictionary containing mappings from an edX IDA to a dictionary of various configuration values (REQUIRED). 
+        The structure of APPS_TO_CONFIG should be
+        
+        APPS_TO_CONFIG:
+            <IDA name>:
+                app_repo: <GitHub repository name for the IDA>
+                app_repo_branch: <branch of the above repository to checkout>
+                config_branch: <GitHub branch of configuration repository to checkout>
+        For example, 
+        APPS_TO_CONFIG:
+            ecommerce:
+                app_repo: 'ecommerce'
+                app_repo_branch: 'master'
+                config_branch: 'master'
+        Note that the default for app_repo_branch is master, and the default for config_branch is master; therefore, they do not need to be specified unless you are providing overrides.
+        
+        For example,
+        APPS_TO_CONFIG:
+            ecommerce:
+                app_repo: 'ecommerce'
+                
+        accomplishes the same as the dictionary above.
+    CONFIGURATION_REPO_URL: URL of the configuration GitHub repository (REQUIRED)
+    EDX_REPO_ROOT: the URL of the edX GitHub organization (REQUIRED)
+    CONFIG_JSON_FILE_CREDENTIAL_ID: ID of the credentials set up in Jenkins to log into DockerHub; defaults to docker-config-json; this refers to the docker-config-json expected credential described below
+    FOLDER_NAME: the name of the folder in which the downstream per-app jobs will live; defaults to DockerCI
+    ACCESS_CONTROL: list of github users or groups who will have access to the jobs
+    Expected credentials - these will normally be set up on the Folder.
+    docker-config-json: a file credential that gives each job in the folder permission to push to DockerHub
+*/
+
+package devops.jobs
+
+import static org.edx.jenkins.dsl.YAMLHelpers.parseYaml
+import org.yaml.snakeyaml.error.YAMLException
+
+import static org.edx.jenkins.dsl.DevopsConstants.common_read_permissions
+import static org.edx.jenkins.dsl.Constants.common_logrotator
+import static org.edx.jenkins.dsl.Constants.common_wrappers
+
+class ConfigurationWatcher {
+    public static def job = { dslFactory, extraVars ->
+        dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/configuration-watcher") {
+
+            assert extraVars.containsKey('CONFIGURATION_REPO_URL') : "Please define CONFIGURATION_REPO_URL"
+            assert extraVars.containsKey('EDX_REPO_ROOT') : "Please define EDX_REPO_ROOT"
+            assert extraVars.containsKey('APPS_TO_CONFIG'): "Please define APPS_TO_CONFIG"
+
+            def config_branch = 'master'
+
+            // to space separated list as string
+            def apps = extraVars.get("APPS_TO_CONFIG").keySet().join(",").replaceAll(',', " ")
```

Comment:
That makes a lot more sense. I'm not entirely sure why I didn't try that, to be honest. I'll do that and try it out to make sure it works.